src/librustc_typeck/collect.rs

```diff
@@ -1940,8 +1940,8 @@ fn get_or_create_type_parameter_def<'a,'tcx>(ccx: &CrateCtxt<'a,'tcx>,
                 lint::builtin::INVALID_TYPE_PARAM_DEFAULT,
                 param.id,
                 param.span,
-                format!("defaults for type parameters are only allowed \
-                         on `struct` or `enum` definitions (see issue #27336)"));
+                format!("defaults for type parameters were only intended \
+                         to be allowed on `struct` or `enum` definitions"));
```

Comment:
`struct` or `enum`?
what about `type`, `trait`, `impl`? They all don't generate the warning currently.